src/plugins/saved_search/server/saved_objects/search.ts

```diff
@@ -67,6 +67,7 @@ export function getSavedSearchObjectType(
           },
         },
         rowsPerPage: { type: 'integer', index: false, doc_values: false },
+        breakdownField: { type: 'keyword', index: false, doc_values: false },
```

Comment:
As a general rule `keyword` fields should have an `ignore_above` value to avoid errors when documents contain larger than expected strings:

```
"reason": "Document contains at least one immense term in field=\"tags\" (whose UTF8 encoding is longer than the max length 32766), all of which were skipped.  Please correct the analyzer to not produce such terms.  The prefix of the first immense term is: '[82, 83, 75, 66, 99, 81, 87, 70, 69, 72, 49, 65, 70, 71, 48, 48, 53, 101, 86, 54, 112, 67, 53, 54, 80, 112, 56, 54, 70, 72]...', original message: bytes can be at most 32766 in length; got 32770",
"caused_by": {
  "type": "max_bytes_length_exceeded_exception",
  "reason": "bytes can be at most 32766 in length; got 32770"
}
```

It might be that `index: false, doc_values: false` avoids this problem altogether but we should test that to be sure. And having the ignore_above already now would be more defensive if we ever decide to query on this field.

Reply:
@rudolf Thanks for the feedback! Any guidance on how to determine what this value should be, or the best approach to test for it -- whichever is the preferred solution here? The value stored here will be a data view name, which from what I can tell also doesn't use `ignore_above`, so I'm not sure what it should be set to: https://github.com/elastic/kibana/blob/main/src/plugins/data_views/server/saved_objects/data_views.ts#L41.

cc @mattkime to keep me honest about data views.

Reply:
Here's the ES field name limit - https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-settings-limit.html

I'd avoid adding `ignore_above` if possible.

Reply:
@rudolf Looking into this more, it's not clear to me what value would make sense for `ignore_above` here. The maximum length of this field would presumably match `index.mapping.field_name_length.limit` since it determines the max length of an ES field name (the value which will be stored in `breakdownField`), but I'm not sure that would even be possible since the setting is configurable. I looked for guidance on how to proceed from the ES team, but even the use of `ignore_above` for this purpose was considered questionable (https://elastic.slack.com/archives/C0D8ST60Y/p1669213401793589). Would failing to save the search due to an excessively long `breakdownField` name be considered incorrect behaviour? Overall I don't have strong opinions on this, but I would like to find a path forward to resolve this blocker.

Reply:
@rudolf since the field isn't indexed, it [_typically_](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-index.html#mapping-index) can't be queried anyway, so I don't think we need to worry about specifying an `ignore_above` value for this particular field right now. Taking a look at the other SO properties here, `ignore_above` isn't specified _for any field_, so at least the mapping's consistent with the other keyword properties.

However, given that elasticsearch [defaults](elastic.co/guide/en/elasticsearch/reference/current/mapping-settings-limit.html#:~:text=Default%20is%2010000%20.&text=Setting%20for%20the%20maximum%20length,to%20limit%20the%20field%20length.) to `Long.MAX_VALUE` as the max field name length limit, maybe using `text` as the mapping type is a better option? @davismcphee is there a technical requirement to have `breakdownField` as `keyword`?

Reply:
@TinaHeiligers Thanks for the input. Using `text` here sounds like it would work to me since we're just storing the field name for the histogram breakdown, but I should check with @kertal to be sure since he added this mapping.

Reply:
> @rudolf since the field isn't indexed, it [_typically_](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-index.html#mapping-index) can't be queried anyway, so I don't think we need to worry about specifying an `ignore_above` value for this particular field right now. Taking a look at the other SO properties here, `ignore_above` isn't specified _for any field_, so at least the mapping's consistent with the other keyword properties.

yes, exactly

But also using `text` should instead of `keyboard` should be fine. let's move forward with `text`.

Reply:
++ `text` seems to be a better match for the data we're putting into this field. Ideally we shouldn't add mappings for fields we don't query on but at the moment this is the only validation for many analytics saved object types. Once there's an API layer with validation these fields can/should be removed.

Reply:
Thanks all for the feedback! I updated the mapping to use `text` instead so I think we're all good here now.